pandas/tests/series/test_apply.py

```diff
@@ -164,6 +164,15 @@ def test_apply_dict_depr(self):
         with tm.assert_produces_warning(FutureWarning):
             tsdf.A.agg({'foo': ['sum', 'mean']})
 
+    def test_apply_categorical_with_nan_values(self):
+        # GH 20714
+        s1 = pd.Series(['1-1', '1-1', np.NaN], dtype='category')
+        s1 = s1.apply(lambda x: x.split('-')[0])
+
+        s2 = pd.Series(['1', '1', np.NaN], dtype='category')
```

Comment:
Can you use the `result`/`expected` variable naming convention for this test, e.g.

```python
s = pd.Series(['1-1', '1-1', np.NaN], dtype='category')
result = s.apply(lambda x: x.split('-')[0])
expected = pd.Series(['1', '1', np.NaN], dtype='category')
```

Additionally, the second example in the issue is the one that needs to be tested: `pd.Series(['1-1','1-2',np.NaN], dtype='category')`.  The example you're testing is already working, so it doesn't test if the bug was actually fixed.  Perhaps you could parametrize over the values of the `Series` and test both since `expected` is the same in both cases, e.g.

```python
@pytest.mark.parametrize('values', [
    ['1-1', '1-1', np.NaN],
    ['1-1', '1-2', np.NaN]])
def test_apply_categorical_with_nan_values(self, values):
    # GH 20714
    s = pd.Series(values, dtype='category')
    ...
```

Reply:
Thanks for the indications @jschendel !

Anyway I found another problem, doing:
```python
@pytest.mark.parametrize('values', [
    ['1-1', '1-1', np.NaN],
    ['1-1', '1-2', np.NaN]])
def test_apply_categorical_with_nan_values(self, values):
s = pd.Series(values, dtype='category')
result = s.apply(lambda x: x.split('-')[0])
expected = pd.Series(['1', '1', np.NaN], dtype='category')
tm.assert_series_equal(result, expected, check_dtype=False)
```
I get an error due to the different classes of ```result``` and ```expected```:
```
AssertionError: Categorical Expected type <class 'pandas.core.arrays.categorical.Categorical'>, found <class 'numpy.ndarray'> instead
```
This situation was already commented on issue #20714 and remarked as out of scope for this PR. 
Knowing that, this test should pass. I thought I could do ```.astype('category')``` on ```result```, but maybe I'm just missing another assert function inside tm that will work for this. Any ideas?